File: data/compilers_comparison/src/examples/second_example/src/func44.c
#include "second_example.h" 
array_t* func44(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array52;
      if (pCounter > 0) {
         array52 = vars->data[--pCounter];
         array52->refC++;
         DEBUG_COPY(array52->id);
      } else {
         array52 = (array_t*)malloc(sizeof(array_t));
         array52->size = 121;
         array52->refC = 1;
         array52->id = 52;
         array52->data = (unsigned int*)malloc(array52->size*sizeof(unsigned int));
         memset(array52->data, 0, array52->size*sizeof(unsigned int));
         DEBUG_NEW(array52->id);
      }
      unsigned int loop18 = 0;
      unsigned int loopLimit18 = (rand()%loopsFactor)/4 + 1;
      for(; loop18 < loopLimit18; loop18++) {
         for (int i = 0; i < array52->size; i++) {
            array52->data[i]--;
         }
         for (int i = 0; i < array52->size; i++) {
            if (array52->data[i] == 29) { 
               return array52;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array52;
      array_t* array53 = func54(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array53->id);
      free(params0.data);
      array53->refC--;
      if(array53->refC == 0) {
         free(array53->data);
         free(array53);
         DEBUG_FREE(array53->id);
      }
      array52->refC--;
      if(array52->refC == 0) {
         free(array52->data);
         free(array52);
         DEBUG_FREE(array52->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array54 = func50(&params0, loopsFactor);
      DEBUG_RETURN(array54->id);
      free(params0.data);
      array54->refC--;
      if(array54->refC == 0) {
         free(array54->data);
         free(array54);
         DEBUG_FREE(array54->id);
      }
   }
   array_t* array55;
   if (pCounter > 0) {
      array55 = vars->data[--pCounter];
      array55->refC++;
      DEBUG_COPY(array55->id);
   } else {
      array55 = (array_t*)malloc(sizeof(array_t));
      array55->size = 428;
      array55->refC = 1;
      array55->id = 55;
      array55->data = (unsigned int*)malloc(array55->size*sizeof(unsigned int));
      memset(array55->data, 0, array55->size*sizeof(unsigned int));
      DEBUG_NEW(array55->id);
   }
   return array55;
}

File: data/multilang_comparison/program_nim/src/func44.nim
proc func44*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var array81: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array81 = vars.data[pCounter]
         inc array81.refC
         debugCopy(array81.id)
      else:
         array81 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array81.size = 772
         array81.refC = 1
         array81.id = 81
         array81.data = cast[ptr UncheckedArray[uint32]](alloc(array81.size * sizeof(uint32)))
         zeroMem(array81.data, array81.size * sizeof(uint32))
         debugNew(array81.id)
      let loopLimit27 = (50) div (4) + 1
      for loop27 in 0 ..< loopLimit27:
         discard
         for i in 0 ..< array81.size:
            dec array81.data[i]
         for i in 0 ..< array81.size:
            if array81.data[i] == 90:
               return array81
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array81
      var array82: ptr ArrayT = func54(addr params0, getPath(), loopsFactor)
      debugReturn(array82.id)
      dealloc(params0.data)
      dec array82.refC
      if array82.refC == 0:
         dealloc(array82.data)
         dealloc(array82)
         debugFree(array82.id)
      dec array81.refC
      if array81.refC == 0:
         dealloc(array81.data)
         dealloc(array81)
         debugFree(array81.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array83: ptr ArrayT = func50(addr params0, getPath(), loopsFactor)
      debugReturn(array83.id)
      dealloc(params0.data)
      dec array83.refC
      if array83.refC == 0:
         dealloc(array83.data)
         dealloc(array83)
         debugFree(array83.id)
   var array84: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array84 = vars.data[pCounter]
      inc array84.refC
      debugCopy(array84.id)
   else:
      array84 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array84.size = 669
      array84.refC = 1
      array84.id = 84
      array84.data = cast[ptr UncheckedArray[uint32]](alloc(array84.size * sizeof(uint32)))
      zeroMem(array84.data, array84.size * sizeof(uint32))
      debugNew(array84.id)
   return array84

